plugins/woocommerce/src/Internal/Admin/Settings.php

```diff
@@ -240,7 +240,19 @@ public function add_component_settings( $settings ) {
 
 		$settings['isWooPayEligible'] = WCPayPromotionInit::is_woopay_eligible();
 
-		$settings['gutenberg_version'] = defined( 'GUTENBERG_VERSION' ) ? constant( 'GUTENBERG_VERSION' ) : 0;
+		$has_gutenberg             = is_plugin_active( 'gutenberg/gutenberg.php' );
+		$gutenberg_version = '';
+		if ( $has_gutenberg ) {
+			if ( defined( 'GUTENBERG_VERSION' ) ) {
+				$gutenberg_version = GUTENBERG_VERSION;
+			}
+
+			if ( ! $gutenberg_version ) {
+				$gutenberg_data    = get_plugin_data( WP_PLUGIN_DIR . '/gutenberg/gutenberg.php' );
+				$gutenberg_version = $gutenberg_data['Version'];
+			}
+		}
+		$settings['gutenberg_version'] = $has_gutenberg ? $gutenberg_version : 0;
```

Comment:
@woocommerce/ghidorah, in particular @moon0326 I updated this code to be more robust. When I was testing this locally `GUTENBERG_VERSION` did not exist and the version always came back as `0`, the above seemed to fix this.